webservices/common/util.py

```diff
@@ -45,3 +47,47 @@ def get_class_by_tablename(tablename):
     for c in db.Model._decl_class_registry.values():
         if hasattr(c, '_sa_class_manager') and c._sa_class_manager.mapper.mapped_table == tablename:
             return c
+
+
+
+def year_defaults(**kwargs):
+    """ We have two ways of picking time and both are now optional.
+
+    It would be nice to deprecate cycle for itemized endpoints that
+    have receipt date in the future, but we can keep it going now for
+    a smoother customer experience and try infer the most consistent
+    defaults.
+
+    Here are the conditions:
+        - if no date supplied default to the current 2-year period
+        - if a cycle is selected, that will provide sufficient date filtering
+        - if there is a min and max date check to make sure there isn't more than a 6 year period
+        - if cycle and only one min date or max-date, it adds the restriction with in the 2-year period
+    """
+    min_date = kwargs.get('min_date')
+    max_date = kwargs.get('max_date')
+    cycle = kwargs.get('two_year_transaction_period')
+
+    if min_date is None and max_date is None and cycle is None:
+        kwargs['two_year_transaction_period'] = CURRENT_CYCLE
+        return kwargs
+    if cycle is not None:
+        return kwargs
+    # this is for cases where only one date (min_date or max_date) is present
+    if not min_date:
```

Comment:
This conditional may need to be more explicit, e.g., `if not min_date and max_date` to make sure we capture the case correctly.